src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -97,17 +106,18 @@ void PitchShiftEffect::processChannel(
             pInput,
             engineParameters.framesPerBuffer());
     pState->m_pRubberBand->process(
-            //static_cast<const float* const*>(pState->m_retrieveBuffer),
             pState->m_retrieveBuffer,
             engineParameters.framesPerBuffer(),
             false);
 
+    // Sets number of samples for current RubberBand process latency.
+    setGroupDelay(pState->m_pRubberBand->getLatency() * engineParameters.channelCount());
```

Comment:
I think it would be better to format this in frames.
Can this ever be a float? Can we put the unit into the function name?

Reply:
> I think it would be better to format this in frames.

I think it could be maybe possible. Anyway, I would like to mention a few reasons, why the samples option was used.

First is, that the EngineEffectsDelay is based on the [EngineFilterDelay,](https://github.com/mixxxdj/mixxx/blob/935ba94f7e498eebd7c3710b2f0df35d989d9a7d/src/engine/filters/enginefilterdelay.h#L34) which uses samples and the thought was to keep the structure as similar as possible.

The second reason was, that for now the stereo signal is kept as default. If I would like to keep the other possible options in mind for maybe some extensions in the future, in the EngineEffectChain it isn't possible to get the number of channels, at least not straight. The information on the number of channels is possible to get via EngineParameters ([created later](https://github.com/mixxxdj/mixxx/blob/935ba94f7e498eebd7c3710b2f0df35d989d9a7d/src/engine/effects/engineeffect.cpp#L181) through process() method call in EngineEffectChain).

And at least the last third and the main reason based on the second one, in the EngineEffectsDelay process() method, it is necessary to work with samples in the work with audio buffers, not with the frames. It can be used frames if it would be possible in EngineEffectsDelay to get information about the number of channels of input audio signal.

So, I would like to summarize the main idea. It is possible if I can get information about the number of channels in the EngineEffectChain process() method, or at least in the EngineEffectsDelay process() method.

> Can this ever be a float?

If I assume, that it will be worked with the samples or frames, it isn't possible. It can be possible if we thought of to option to work with the delay in milliseconds, but IMO it doesn't make much sense. If I'm wrong, please correct me.

> Can we put the unit into the function name?

Yeah, of course.

Reply:
Ok, that sounds reasonable. 
If we put an assertion that the stereo signal is it swapped it works for me as well.

Reply:
Okay, thank you. Good idea. Still, I see now on my solution and proposal of using samples rather than frames maybe a little problem, that if it will be worked in the future with other numbers of channels than the stereo signal, it would be harder to assert the invalid number of samples. To check the invalid value the number of samples has to be divisible by the number of channels. And because of that, the information about the number of channels has to be accessible in the EngineEffecChannel or EngineEffectsDelay, so this situation isn't still solved.